### ISSUE TITLE
Install Error with requirements.txt

### ISSUE THREAD

**Issue by r530044129** [alexzhou907/DreamPropeller#1]
I install ninja and torch2.1.0+cu121 first, then install the other requirement, but when i run the code, i meet the error
pip install -r requirements.txt
...
          _build_ext.run(self)
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\setuptools\_distutils\command\build_ext.py", line 345, in run
          self.build_extensions()
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\torch\utils\cpp_extension.py", line 873, in build_extensions
          build_ext.build_extensions(self)
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\setuptools\_distutils\command\build_ext.py", line 467, in build_extensions
          self._build_extensions_serial()
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\setuptools\_distutils\command\build_ext.py", line 493, in _build_extensions_serial
          self.build_extension(ext)
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\setuptools\command\build_ext.py", line 246, in build_extension
          _build_ext.build_extension(self, ext)
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\setuptools\_distutils\command\build_ext.py", line 548, in build_extension
          objects = self.compiler.compile(
                    ^^^^^^^^^^^^^^^^^^^^^^
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\torch\utils\cpp_extension.py", line 845, in win_wrap_ninja_compile
          _write_ninja_file_and_compile_objects(
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\torch\utils\cpp_extension.py", line 1774, in _write_ninja_file_and_compile_objects
          _run_ninja_build(
        File "C:\ProgramData\anaconda3\envs\dreampropeller\Lib\site-packages\torch\utils\cpp_extension.py", line 2116, in _run_ninja_build
          raise RuntimeError(message) from e
      RuntimeError: Error compiling objects for extension
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tinycudann
  Running setup.py clean for tinycudann
  Building wheel for nvdiffrast (setup.py) ... done
  Created wheel for nvdiffrast: filename=nvdiffrast-0.3.1-py3-none-any.whl size=141467 sha256=2bf37bd56803a0c18abb8dab0f879590bee22e5b00d72af419de8cab10d811c3
  Stored in directory: C:\Users\renjiezhang\AppData\Local\Temp\pip-ephem-wheel-cache-wvd3316i\wheels\1f\02\42\f8a0a54a25f789f27dd0a5509a22236057abd5d75f3f69969f
  Building wheel for pysdf (setup.py) ... done
  Created wheel for pysdf: filename=pysdf-0.1.9-cp311-cp311-win_amd64.whl size=128067 sha256=0eadd7f7f7d83d3fde0bccfa948bcbea3ea80b2d266d2467b7fba7a6710cc613
  Stored in directory: c:\users\renjiezhang\appdata\local\pip\cache\wheels\70\14\13\db6ce1ad9f1f170ef0aaeadc7fc58e082a5147a5953e93628a
  Building wheel for envlight (setup.py) ... done
  Created wheel for envlight: filename=envlight-0.1.0-py3-none-any.whl size=40061 sha256=bbf3cd529c7c2e4b46e6689cf2777b4fd3feecd500046577900e98feb8c99fbf
  Stored in directory: C:\Users\renjiezhang\AppData\Local\Temp\pip-ephem-wheel-cache-wvd3316i\wheels\9e\aa\90\b01e0473e0659bde0d0766c6f80e2fdb4d092103587453dcb7
  Building wheel for clip (setup.py) ... done
  Created wheel for clip: filename=clip-1.0-py3-none-any.whl size=1369578 sha256=4f7fdf53ac708fc973c4c9a4bccdf09bf4138d426800bd04acc9f7d858c1cdf7
  Stored in directory: C:\Users\renjiezhang\AppData\Local\Temp\pip-ephem-wheel-cache-wvd3316i\wheels\3f\7c\a4\9b490845988bf7a4db33674d52f709f088f64392063872eb9a
  Building wheel for controlnet_aux (setup.py) ... done
  Created wheel for controlnet_aux: filename=controlnet_aux-0.0.7-py3-none-any.whl size=274369 sha256=220b33f6f9a237ab2ed6720e9732c17b31c62e167eb7817eedf45ae9d25b84a5
  Stored in directory: c:\users\renjiezhang\appdata\local\pip\cache\wheels\08\a1\0f\79d1529bbc60d1598da66052a2c60a2a13e5ac462b5f990653
  Building wheel for ffmpy (setup.py) ... done
  Created wheel for ffmpy: filename=ffmpy-0.3.1-py3-none-any.whl size=5601 sha256=7734b4025ec2dae682138fabf0b312875ebc016deccfc74a51bf3bec4939e92c
  Stored in directory: c:\users\renjiezhang\appdata\local\pip\cache\wheels\99\3b\84\22ac1eab7a10222ac6bbc3f7e69b04f3980db328978c533a3f
  Building wheel for pycollada (setup.py) ... done
  Created wheel for pycollada: filename=pycollada-0.7.2-py3-none-any.whl size=127024 sha256=d1e9a77ad169356ece433027c2985f44b4dba1b487445251d49398f622644c2c
  Stored in directory: c:\users\renjiezhang\appdata\local\pip\cache\wheels\07\31\f9\395db362d2ceb6314508ed93426113131978eaca551cb359eb
Successfully built antlr4-python3-runtime nerfacc nvdiffrast pysdf envlight clip controlnet_aux ffmpy pycollada
Failed to build tinycudann
ERROR: Could not build wheels for tinycudann, which is required to install pyproject.toml-based projects

Maybe it has some relation with my env or something, could you please give us some threestudio custom Extensions? just like those:
https://threestudio-project.github.io/threestudio-extensions/
It will be more convenient!

**Comment by alexzhou907** [comment]
We have only tested on Linux systems. I'm not so sure about Windows. It seems like tinycudann does not support Windows?

**Comment by r530044129** [comment]
yeah, i've met those problem before, it's troublesome to deal with those problem in win
